test/dom/propertyShadowing.ts

```diff
@@ -22,6 +28,29 @@ describe('dom.Struct property shadowing', () => {
         assert.equal(s.get('fieldNames', 1)!.stringValue(), 'cat');
         assert.equal(s.get('fieldNames', 2)!.stringValue(), 'mouse');
 
+        // deleteProperty proxy method
+        assert.isTrue(delete s['greetings']);
+        assert.deepEqual(s.fieldNames(), ['getType', 'getAnnotations', 'fieldNames', 'toString', 'age']);
+
+        // deleteField Struct method
+        assert.isTrue(s.deleteField('age'));
+        assert.deepEqual(s.fieldNames(), ['getType', 'getAnnotations', 'fieldNames', 'toString']);
+
+        // delete for properties that match built-in
+        assert.equal(s.get('toString')!.numberValue(), 10);
+        assert.isTrue(delete s['toString']);
+        assert.equal(s.toString(),'{getType: baz, getAnnotations: 56, fieldNames: [dog, cat, mouse]}');
+        assert.deepEqual(s.fieldNames(), ['getType', 'getAnnotations', 'fieldNames']);
+
+        // delete for field that doesn't exist
+        assert.isFalse(s.deleteField('toString'));
+        assert.isFalse(s.deleteField('greetings'));
+        assert.isFalse(s.deleteField('name'));
+        assert.isUndefined(s['greetings']);
+
+        // deleteField will throw an error if it's called on a dom.Value that isn't a struct
+        assert.throws(() => l.deleteField("1"), Error);
```

Comment:
Since the list `l` is only used in this one check, could you move its definition down to just below the comment?

```suggestion
        // deleteField will throw an error if it's called on a dom.Value that isn't a struct
        let l = dom.Value.from([1, 2, 3]);
        assert.throws(() => l.deleteField("1"), Error);
```